README
This directory tree holds version 1.8.0 of the m17n database.
Copyright (C) 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012,
  2013, 2014
  National Institute of Advanced Industrial Science and Technology (AIST)
  Registration Number H15PRO112
Copyright (C) 2013, 2014, 2015, 2016, 2017 Kenichi Handa <handa@gnu.org>
See the end for copying conditions.

----------------------------------------------------------------------
## INSTALL ##

(0) Prerequisite

The make procedure requires "charmap" files.  The configure script
cheks the directory /usr/share/i18n/charmaps which usually should be
found on GNU/Linux system.

If you don't have that directory, you must get the files from the
source of glibc, and inform the configure script of the directory by
the argument "--with-charmaps=DIR".

If you have "wget", the script "./get-glibc.sh" is a convenient tool
to download the source of glibc (2.3) and extract files in a directory
where the configure script finds automatically, thus no need of
"--with-charmaps=DIR" argument.

Please note that /usr/share/i18n/charmaps may contain incomplete
GB18030.gz (e.g. /xfe/xa0.. /xfe/xfe are not mapped).  In that case,
you need the newer glibc source (glibc-2.3) by using "./get-glibc.sh".

(1) Installation from CVS working directory.

Just after you checked out or updated this package, run the script
"./bootstrap.sh", then follows (2).

(2) Installation from a tar ball.

% ./configure
% make
% make install

----------------------------------------------------------------------
## Files and directories ##

mdb.dir -- defining database files
FLT/ -- directory containing font layout table (*.flt)
LANGDATA/ -- directory containing language names data (*.lnm)
MIM/ -- directory containing input methdos (*.mim)
icons/ -- directory containing icon files of input methdos (*.png)
po/ -- directory containing localized messages of input methods
*.map -- Charaset set mapping tables

----------------------------------------------------------------------
## Thanks ##

We appreciate all contributers (see the section "Contributed files"
below) improving the m17n library.  We also appreciate Harshula
Jayasuriya helping this project in various aspects (setting up
Savannah infrastructure, advising on GIT, appropriate suggestions,
etc).

----------------------------------------------------------------------
## Copyright information ##

Most files in this package are copyrighted by AIST as below;
exceptions are listed in the file AUTHORS.

Copyright (C) 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012,
  2013, 2014
  National Institute of Advanced Industrial Science and Technology (AIST)
  Registration Number H15PRO112
Copyright (C) 2013, 2014, 2015, 2016, 2017 Kenichi Handa <handa@gnu.org>

This file is part of the m17n database; a sub-part of the m17n
library.

The m17n library is free software; you can redistribute it and/or
modify it under the terms of the GNU Lesser General Public License
as published by the Free Software Foundation; either version 2.1 of
the License, or (at your option) any later version.

The m17n library is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public
License along with the m17n library; if not, write to the Free
Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
Boston, MA 02110-1301, USA.


NEWS
* NEWS -- What's new in the m17n database.			-*- outline -*-
Copyright (C) 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013
  2014
  National Institute of Advanced Industrial Science and Technology (AIST)
  Registration Number H15PRO112
See the end for copying conditions.
Copyright (C) 2015, 2016, 2017, 2018 Kenichi Handa <handa@gnu.org>


* Changes in the m17n database 1.8.0

** New input methods are added.

mr-remington.mim: Marathi input method for remington layout
mr-typewriter.mim: Marathi input method for typewriter layout
ne-rom-translit.mim: Nepali input method with roman transliteration
ne-trad.mim: Nepali input method for traditional layout
kn-optitransv2.mim: Kannada input method by the OPTITRANS transliteration system
hi-optitransv2.mim: Hindi input method  by the OPTITRANS transliteration system
ml-enhanced-inscript.mim: Malayalam input method for enhanced inscript layout

** This input method is largely augmented.

lsymbol.mim: Add many symbols.

** Names of these input methods are changed.

uk-kbd.mim: renamed from ua-kbd to uk-kbd
zh-zhuyin.mim: renamed from zh-bopomofo to zh-zhuyin


* Changes in the m17n database 1.7.0

** New input methods are added.

ar-translit.mim: Arabic input method based on Roman transliteration.

latn1-pre.mim: Latin input method simulating "US International Keyboard".

ssymbol.mim, lsymbol.mim: Input method for various symbols.

** A new global variable is added.

fallback-input-method: Comma separated fallback input method names.


* Changes in the m17n database 1.6.5

** All input methods in the m17n contrib package are merged.

** New input methods are added.

sa-itrans.mim: Sanskrit input method with ITRANS method

ml-remington.mim: Malayalam input method.


* Changes in the m17n database 1.6.4

This release is just for bug fixing.


* Changes in the m17n database 1.6.3

** A new input method is added.

ua-kbd.mim -- Ukrainian input method with Ukrainian/Russian keyboard layout

** An input method is deleted.

bo-ewts.mim is moved to m17n-contrib package.


* Changes in the m17n database 1.6.2

** New FLT file.

FLT/HEBR-OTF.flt 


* Changes in the m17n database 1.6.1

** This release is just for a minor bug fix.


* Changes in the m17n database 1.6.0

** An input method is deleted.

si-wijesekera-preedit.mim

** New FLT files are added.

FLT/MLM2-OTF.flt, FLT/KND2-OTF.flt, FLT/TEL2-OTF.flt,
FLT/TML2-OTF.flt, FLT/ORY2-OTF.flt, FLT/GJR2-OTF.flt,
FLT/GUR2-OTF.flt, FLT/BNG2-OTF.flt, FLT/DEV2-OTF.flt,
FLT/MYMR-SIL.flt


* Changes in the m17n database 1.5.5

** An input method is deleted.

si-wijesekera-preedit-alt.mim

** These Vietnamese input methods adopt smart-quotes.

vi-telex.mim, vi-tcvn.mim, vi-vni.mim


* Changes in the m17n database 1.5.4

** A new input method is added.

ta-lk-renganathan.mim (Tamil)


* Changes in the m17n database 1.5.3

This release is just for bug fixing.


* Changes in the m17n database 1.5.2

** A new input method is added.

sa-harvard-kyoto.mim (Sanscrit)


* Changes in the m17n database 1.5.1

** A new FLT is added.

CHAM-GENERIC.flt (For Cham script)

** Several new input methods are added.

vi-tcvn.mim (Vietnamese)
cmc-kbd.mim (Cham)
sa-harvard-kyoto.mim (Sanscrit)


* Changes in the m17n database 1.5.0

** A new FLT is added.

COMBINING.flt (For combining diacritical marks)


* Changes in the m17n database 1.4.0

** Several new input methods are added.

da-post.mim (Danish)
bo-ewts.mim (Tibetan)
fr-azerty.mim (French)

** New database for language names.


* Changes in the m17n database 1.3.4

** Several new input methods are added.

grc-mizuochi.mim (Greek)
ja-trycode.mim (Japanese)
ru-phonetic.mim (Russian)
si-wijesekera-preedit.mim (Sinhala)
th-kesmanee-2.mim, th-pattachote.mim th-pattachote-2.mim (Thai)
th-tis820.mim th-tis820-2.mim (Thai)
ug-kbd.mim (Uyghur)
zh-bopomofo.mim (Chinese bopomofo, renamed from bopo-kbd.mim)

** Several new FLTs are added.

LAOO-OTF.flt (Lao)
THAI-OTF.flt (Thai)

** Gettextized.


* Changes in the m17n database 1.3.0

** Wildcard handling is added to mdb.dir.

For instance, the current mdb.dir file contains this single line:
  (input-method * "*.mim")
instead of lines for each input methods.

** Several new input methods are added.

bn-unijoy.mim (Bengali)
hi-typewriter.mim (Hindi)
lo-kbd.mim (Lao)
si-wijesekera.mim (Sinhala)
sv-post.mim (Swedish)
vi-vni.mim (Vietnamese)
zh-cangjie.mim, zh-quick.mim (Chinese)
zh-py-b5.mim, zh-py-gb.mim (Chinese, restricted version of zh-py.mim)
zh-tonepy-b5.mim, zh-tonepy-gb.mim (Chinese, restricted version of zh-tone.mim)

** Several new FLTs are added.

ARAB-OTF-NO-GPOS.flt (Arabic)
KHMR-OTF.flt (Khmer)
LAOO-GENERIC.flt (Lao)
MLYM-RACHANA.flt (Malayalam)
THAI-GENERIC.flt (Thai)
TIBT-OTF.flt (Tibetan)


* Changes in the m17n database 1.2.0

** Naming convention for *.flt and *.mim is changed.

Script names : ISO 15924 (four letter codes)
Language names : ISO 639-1 (two letter codes)
See ChangeLog for details.

** Several new input methods are added.

dv-phonetic.mim (Thaana script)
hy-kbd.mim (Armenian script)
ko-romaja.mim (Korean Hangul script)
my-kbd.mim (Myanmar script)
si-samanala.mim (Sinhala script)
zh-pinyin.min (PinYin script)
bopo-kbd.mim (Bopomofo script)
syrc-phonetic.mim (Syriac script)
latn-pre.mim (Latin script)

** Several new FLTs are added.

SINH-OTF.flt (Sinhala script)
SYRC-OTF.flt (Syriac script)

** Several new char-tables are added.

BLOCK.tab (Unicode Blocks)
CASE-MAPPING.tab CASED.tab SOFT-DOTTED.tab (for case conversion)
LINEBREAK.tab (for linebreaking)
LANGUAGE.tbl (ISO639-1, ISO639-2 language codes)
SCRIPT-LANGUAGE.tbl (script vs languages)

** A new fontset generic.fst is added.


* Changes in the m17n database 1.1.0

** Several new FLTs are added.

ARABIC-OTF.flt (Arabic script)
BEN-OTF.flt (Bengali script)
GUJ-OTF.flt (Gujarati script)
GUR-OTF.flt (Gurmukhi script)
KAN-OTF.flt (Kannada script)
MYANMAR-ZEDI.flt (Myanmar script)
ORI-OTF.flt (Oriya script)

LANGNAME.awk
# LANGNAME.awk -- awk script to generate LANGNAME.en and LANGNAME.fr
# Copyright (C) 2007
#   National Institute of Advanced Industrial Science and Technology (AIST)
#   Registration Number H15PRO112

# This file is part of the m17n database; a sub-part of the m17n
# library.

# The m17n library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public License
# as published by the Free Software Foundation; either version 2.1 of
# the License, or (at your option) any later version.

# The m17n library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# Lesser General Public License for more details.

# You should have received a copy of the GNU Lesser General Public
# License along with the m17n library; if not, write to the Free
# Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
# Boston, MA 02110-1301, USA.

BEGIN {
    if (LANG == "en") {
	Language = "English";
	LANGUAGE = "ENGLISH";
    } else {
	Language = "French";
	LANGUAGE = "FRENCH";
    }
    printf ";; LANGNAME.%s -- %s language names	-*- mode:lisp; coding:utf-8; -*-\n", LANG, Language;
    print ";; Copyright (C) 2007";
    print ";;   National Institute of Advanced Industrial Science and Technology (AIST)";
    print ";;   Registration Number H15PRO112";
    print "";
    print ";; This file is part of the m17n database; a sub-part of the m17n";
    print ";; library.";
    print "";
    print ";; The m17n library is free software; you can redistribute it and/or";
    print ";; modify it under the terms of the GNU Lesser General Public License";
    print ";; as published by the Free Software Foundation; either version 2.1 of";
    print ";; the License, or (at your option) any later version.";
    print "";
    print ";; The m17n library is distributed in the hope that it will be useful,";
    print ";; but WITHOUT ANY WARRANTY; without even the implied warranty of";
    print ";; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU";
    print ";; Lesser General Public License for more details.";
    print "";
    print ";; You should have received a copy of the GNU Lesser General Public";
    print ";; License along with the m17n library; if not, write to the Free";
    print ";; Software Foundation, Inc., 51 Franklin Street, Fifth Floor,";
    print ";; Boston, MA 02110-1301, USA.";
    print ";;";
    print ";; Each line of this file has this form:";
    printf ";;    (ISO639-2 \"%s-NAME\" ...)\n", LANGUAGE;
    print ";; ISO639-2: 3-letter language code of ISO639-2.";
    printf ";; %s-NAME: %s name of the language\n", LANGUAGE, Language;
    print ";;    in ISO639-2 with the following modifications:";
    if (LANG == "en") {
	print ";;	'Greek, Modern (1453-)' is changed to 'Greek'";
	print ";;	Tailing ' (Other)' is deleted.";
	print ";;	Tailing ' languages' is deleted.";
	print ";;	Tailing part specifying the date (e.g. ' (ca.450-1100)') is deleted.";
	print ";;	Reorder, for instance, 'English, Old' to 'Old English'.";
    } else {
	print ";;	'grec moderne (après 1453)' is changed to 'grec'";
	print ";;	Tailing ', autres langues' is deleted.";
	print ";;	Tailing ', autres' is deleted.";
	print ";;	Tailing ', langues' is deleted.";
	print ";; 	Tailing ', langues (famille)' is deleted.";
	print ";;	Tailing part specifying the date (e.g. ' (ca.450-1100)') is deleted.";
	print ";;	Reorder, for instance, 'saxon, bas' to 'bas saxon'.";
    }
    print ";;	Delete duplicated names after the above modifications.";
    print "";
    printf "(language name %s)\n", LANG;
    print "";
}

{
    iso = $1;
    for (idx in names) delete names[idx];
    if (LANG == "en") {
	split($4, array, "; ");
	for (idx in array) {
	    name = array[idx]
	    if (name == "Greek, Modern (1453-)")
		name = "Greek";
	    else if (name == "Multiple languages")
		;
	    else {
		gsub(" \\(Other\\)$", "", name);
		gsub(" languages$", "", name);
		if (name ~ /\(.*[0-9].*\)$/) {
		    gsub(" \\([^)]*\\)$", "", name);
		}
		if (name ~ ", ") {
		    split(name,array2,", ");
		    name = array2[2] " " array2[1];
		}
	    }
	    names[name] = 1;
	}
    } else {
	split($5, array, "; ");
	for (idx in array) {
	    name = array[idx]
	    if (name == "grec ancien (jusqu'à 1453)")
		name = "grec";
	    else {
		gsub(", autres langues$", "", name);
		gsub(", autres$", "", name);
		gsub(", langues$", "", name);
		gsub(", langues \\(famille\\)$", "", name);
		if (name ~ /\(.*[0-9].*\)$/)
		    gsub(" \\([^)]*\\)$", "", name);
		if (name ~ ", ") {
		    split(name,array2,", ");
		    name = array2[2] " " array2[1];
		}
	    }
	    names[name] = 1;
	}
    }
    printf "(%s", $1;
    for (name in names) {
	printf " \"%s\"", name;
    }
    printf ")\n";
}

# Local Variables:
# coding: utf-8
# End:
